app/lib/reports/pages/reports.dart

```diff
@@ -1,12 +1,160 @@
+      child: BlocBuilder<ReportsCubit, ReportsState>(
+        builder: (context, state) {
+          return RoundedCard(
+            child: Padding(
+              padding: const EdgeInsets.fromLTRB(4, 8, 4, 0),
+              child: Column(children: [
+                Card(
+                  shape: RoundedRectangleBorder(
+                    borderRadius: BorderRadius.circular(16),
+                  ),
+                  color: PharmeTheme.secondaryColor,
+                  child: Padding(
+                    padding: const EdgeInsets.all(8),
+                    child: Column(children: [
+                      Text(
+                        context.l10n.reports_page_disclaimer_title,
+                        style: PharmeTheme.textTheme.headline6!
+                            .copyWith(color: Colors.white),
+                      ),
+                      SizedBox(height: 6),
+                      Row(
+                        children: [
+                          Flexible(
+                            child: Text(
+                              context.l10n.reports_page_disclaimer_text,
+                              style: PharmeTheme.textTheme.bodyMedium!
+                                  .copyWith(color: Colors.white),
+                            ),
+                          ),
+                          SvgPicture.asset('assets/images/reports_icon.svg')
+                        ],
+                      )
+                    ]),
+                  ),
+                ),
+                state.when(
+                  initial: Container.new,
+                  error: () => Text(context.l10n.err_generic),
+                  loading: () => Center(child: CircularProgressIndicator()),
+                  loaded: _buildMedicationsList,
+                )
+              ]),
+            ),
+          );
+        },
+      ),
     );
   }
+
+  Widget _buildMedicationsList(List<MedicationWithGuidelines> medications) {
+    // TODO(kolioOtSofia): filter relevant guidelines and remove medications with ok warning levels only #270
+    return Flexible(
+      child: ListView.separated(
+          itemBuilder: (context, index) {
+            final el = medications[index];
+            return ReportCard(
+              warningLevel:
+                  _convertToWarningLevel(el.guidelines[0].warningLevel),
+              onTap: () {},
+              medicationName: el.name,
+              medicationDescription: el.description,
+            );
+          },
+          separatorBuilder: (_, __) => SizedBox(height: 8),
+          itemCount: medications.length),
+    );
+  }
+
+  WarningLevel _convertToWarningLevel(String? str) {
+    switch (str) {
+      case 'danger':
+        return WarningLevel.danger;
+      case 'warning':
+        return WarningLevel.warning;
+      case 'ok':
+        return WarningLevel.ok;
+      default:
+        throw Exception('Warning level not supperted');
+    }
+  }
+}
+
+class ReportCard extends MedicationCard {
```

Comment:
Flutter conventionally favours composition over inheritance, so I guess instead of inherit from `MedicationCard` you could just return a `MedicationCard` with specific properties.

Reply:
Honestly, there is pretty much nothing that I am taking from the `MedicationCard` so I will just make this an independent classs